docs/markdown/Contributing.md

```diff
@@ -412,23 +414,23 @@ The `skip_on_env` key can be used to specify a list of environment variables. If
 at least one environment variable in the `skip_on_env` list is present, the test
 is skipped.
 
-#### skip_on_jobname
+#### expect_skip_on_jobname
 
-The `skip_on_jobname` key contains a list of strings. If the `MESON_CI_JOBNAME`
+The `expect_skip_on_jobname` key contains a list of strings. If the `MESON_CI_JOBNAME`
 environment variable is set, and any of them are a sub-string of it, the test is
 expected to be skipped (that is, it is expected that the test will output
 `MESON_SKIP_TEST`, because the CI environment is not one in which it can run,
 for whatever reason).
 
-The test is failed if it skips or runs unexpectedly.
+The test is failed if it runs unexpectedly.
```

Comment:
I am pretty sure this was originally correct... if you use (expect_)skip_on_jobname and a CI job that:
- didn't expect to skip, skips, it is an "unexpected skip"
- expected to skip, but runs, it is an "unexpected run"

Reply:
> didn't expect to skip, skips, it is an "unexpected skip"

hm, I think I see what you mean, but that's a property of tests in general, and not specific to this option, no?

Reply:
ah, so that meant "skips unexpectedly or runs unexpectedly".

I'm fine with changing it to that instead

Reply:
I've replaced it with "The test is failed if it either skips unexpectedly or runs unexpectedly."

Reply:
OK, looked a bit more, the unexpected skip part is not true, it's hardcoded and only fail sometimes: https://github.com/mesonbuild/meson/blob/05bbe45cc228c78c47f57c88a04acd937a60f8bb/run_project_tests.py#L996-L1008

Which makes me tend back to my original text again, since unexpected skips don't always fail a test.

Let me know what you prefer.

Reply:
But unexpected skips fail a test under any circumstances where unexpected runs do... the skip_dont_care function defines whether or not meson should *consider* the skippability status as relevant.

What you want is: https://github.com/mesonbuild/meson/blob/05bbe45cc228c78c47f57c88a04acd937a60f8bb/run_project_tests.py#L1307-L1319

i.e. "if we care about skippability, mark it as skip_equals_expectedskip if we ran the test (preconditions met) and skip does, in fact, equal expectedskip"

https://github.com/mesonbuild/meson/blob/05bbe45cc228c78c47f57c88a04acd937a60f8bb/run_project_tests.py#L1332-L1340

and if it didn't match, then:
- error out
- check whether it was skipped. If it was skipped, the error is unexpected-skip, if it was not skipped then the error is unexpected-run

Reply:
To put it more simply, the function "skip_dont_care" determines whether we *expect* either skipping or not-skipping.

For example, in non-CI environments, we don't *expect* anything to either skip or not skip. We don't control the test dependencies, so lots of things may either skip or not skip depending on what is installed. We equally don't care either way. expected-skips define whether skips are expected in named CI environments.

Tests without a test.json don't state whether they skip, so the documentation on the test.json file format isn't relevant. But frameworks without a test.json assume none can skip. Totally unrelated to the test.json documentation.

Reply:
ok, so all the "expected skip" things needs to be prefixed with "only applies to tests in the 'frameworks' directory and only when run in CI"?

At this point I'm just happy if someone tells me what I need to change/revert to get this merged.

(imho the whole expected skip thing is not worth the hassle, as it's not possible to test locally, but that's another issue)

Reply:
It only has an effect when run in CI. That's all.

I think the PR is good as it is -- just checked and it still says "The test is failed if it either skips unexpectedly or runs unexpectedly".

Reply:
> It only has an effect when run in CI. That's all.

And what about this?

https://github.com/mesonbuild/meson/blob/05bbe45cc228c78c47f57c88a04acd937a60f8bb/run_project_tests.py#L1001-L1006

> I think the PR is good as it is -- just checked and it still says "The test is failed if it either skips unexpectedly or runs unexpectedly".

ok, thank you